lib/ops/opsservice/endpoints.go

```diff
@@ -75,30 +75,18 @@ func (o *Operator) GetApplicationEndpoints(key ops.SiteKey) ([]ops.Endpoint, err
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	namespaceList, err := client.CoreV1().Namespaces().List(metav1.ListOptions{})
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
 
 	var endpoints []ops.Endpoint
 	for _, e := range site.app.Manifest.Endpoints {
 		if e.Hidden {
 			continue
 		}
 
-		var serviceList *v1.ServiceList
-		for _, ns := range namespaceList.Items {
-			services, err := client.CoreV1().Services(ns.Name).List(metav1.ListOptions{
-				LabelSelector: utils.MakeSelector(e.Selector).String(),
-			})
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
-			if serviceList == nil {
-				serviceList = services
-			} else {
-				serviceList.Items = append(serviceList.Items, services.Items...)
-			}
+		serviceList, err := client.CoreV1().Services(constants.AllNamespaces).List(metav1.ListOptions{
```

Comment:
I'm not so sure using the empty string will query across all namespaces - correct me if I'm wrong here.

Reply:
Yep, it will: https://github.com/kubernetes/client-go/blob/master/rest/request.go#L451-L453.